### PR TITLE
Revamped search interface to use the support library SearchView

### PR DESCRIPTION
**Pull request by nwalters512**

### CODE REVIEW

**Comment by jonan** [comment]
The new SearchView looks really nice, but I've seen two issues when giving it a try:

- Once in the search activity (after the first search), if you try to search again and you press on a recently search item, the search input doesn't get updated, so you can't see the search text nor search again.
- In b201ba888504b145fa5146cb7ef25151ea71fca4 I added a functionality to allow editing the last search. I find this useful for fixing typos or when adding more search terms. This feature is gone with the new SearchView.

**Comment by nwalters512** [comment]
Good catches. I'll fix those issues.

**Comment by nwalters512** [comment]
That should resolve the remaining issues. In the results activity, the search view is now permanently expanded to make it easy to refine the query.

**Comment by nwalters512** [comment]
Anything else I need to do here?

**Comment by jonan** [comment]
Sorry, I haven't had time to test it yet. I'll try to look at it tonight.

**Comment by jonan** [comment]
I've been testing this and I still have some issues.

 * First, I think you accidentally remove this line when merging: https://github.com/jonan/ForkHub/pull/166/commits/bbc5bc7824c784fdc589767561348a620920099b#diff-dd1e125e3fae1baf04c3bc756062096eL39. Please rebase your branch with current master so we can fix it.

* I find weird that after going to the search view, pressing back sends you to the home activity with the search bar active. I think it should be deactivated once you go to the search view.

* I don't like how the button to create a new issues moves when you press search. Can we hide it or find another solution?

**Comment by nwalters512** [comment]
Those commits should solve your first two bullets. I'm not quite sure what you mean by the third one. In the Repository view on the Issues tab, pressing the search icon will hide the "+" in the toolbar (technically, it's moved to the overflow menu). Are you seeing different behavior?

**Comment by jonan** [comment]
I was, but you seem to have solve it with the new searchView listeners.
Everything looks fine now, I'll do a bit more testing and merge it tonight if nothing comes up.

Many thanks!

**Comment by jonan** [comment]
Sorry, I found one more issue (hopefully the last one).

There's no search button in `IssueSearchActivity`, so after searching for an issue, you can't search again without going back :disappointed: 

BTW, note that you're including some useless `imports` and that we don't indent `case` statements.

**Comment by nwalters512** [comment]
I'll work on those issues.

Is there a way to change how the IDE formats switch statements? I'm using the [official Android code style](https://github.com/android/platform_development/blob/master/ide/intellij/codestyles/AndroidStyle.xml), which indents `case` by defult, and I couldn't find an option in Android Studio to change that behavior.

**Comment by jonan** [comment]
Sure, `Settings > Editor > Code Style > Java > Wrapping and Braces > 'switch' statement`.

File: app/src/main/java/com/github/mobile/ui/issue/IssueSearchActivity.java [hunk 1]
```diff
@@ -50,6 +55,8 @@
 
     private String query;
```
Comment (jonan) [inline app/src/main/java/com/github/mobile/ui/issue/IssueSearchActivity.java]: This variable is useless now

File: app/src/main/java/com/github/mobile/ui/search/SearchActivity.java [hunk 2]
```diff
@@ -105,20 +123,26 @@ protected void onNewIntent(Intent intent) {
...
+            });
+
+            search(query);
+        }
     }
 
     private void search(final String query) {
-        this.query = query;
-        getSupportActionBar().setTitle(query);
+        getIntent().putExtra(QUERY, query);
```
Comment (jonan) [inline app/src/main/java/com/github/mobile/ui/search/SearchActivity.java]: What's the purpose of this?

File: app/src/main/java/com/github/mobile/ui/repo/RepositoryViewActivity.java [hunk 3]
```diff
@@ -158,6 +160,19 @@ public boolean onSearchRequested() {
     }
 
     @Override
+    public void startActivity(Intent intent) {
+        // Inject extra information into search intents
+        // Search intents will probably come from the "Issues" fragment
+        if(Intent.ACTION_SEARCH.equals(intent.getAction())) {
```
Comment (jonan) [inline app/src/main/java/com/github/mobile/ui/repo/RepositoryViewActivity.java]: Space missing after `if` :wink:

**Comment by jonan** [comment]
I've added three very picky comments, the rest is perfect and ready for merge :wink:

File: app/src/main/java/com/github/mobile/ui/search/SearchActivity.java (hunk 2 above)
Comment (nwalters512) [inline app/src/main/java/com/github/mobile/ui/search/SearchActivity.java]: Oops, that was a remnant of when I was approaching search differently. It's unneeded now.

**Comment by nwalters512** [comment]
There you go! Props to you for having the patience to go through everything with such a fine-toothed comb :stuck_out_tongue:

**Comment by jonan** [comment]
Thank you @nwalters512! Expect a new release soon :+1:

**Pull request merged**